code-examples/core_features/web3names/5_web3name.ts

```diff
@@ -0,0 +1,93 @@
+import fetch from 'node-fetch'
+
+import { Attestation, RequestForAttestation } from '@kiltprotocol/core'
+import { DidUtils, Web3Names, resolveDoc } from '@kiltprotocol/did'
+
+import type { IRequestForAttestation } from '@kiltprotocol/types'
+
+// The type to filter the endpoints of the retrieved DID.
+const PUBLISHED_CREDENTIAL_COLLECTION_V1_TYPE =
+  'KiltPublishedCredentialCollectionV1'
+
+const verifyCredential = async (
+  publishedCredential: RequestForAttestation
+): Promise<boolean> => {
+  // Retrieve the on-chain attestation information about the credential.
+  const onChainAttestation = await Attestation.query(
+    publishedCredential.rootHash
+  )
+  if (!onChainAttestation || onChainAttestation.revoked) {
+    return false
+  }
+  // Verify the credential integrity and the subject's signature.
+  return (
+    publishedCredential.verifyData() && publishedCredential.verifySignature()
+  )
+}
+
+export async function main(web3Name: Web3Names.Web3Name) {
+  const didForWeb3Name = await Web3Names.queryDidForWeb3Name(web3Name)
+  if (!didForWeb3Name) {
+    throw `No DID found for "${didForWeb3Name}"`
+  }
+
+  console.log(`DID for "${web3Name}": ${didForWeb3Name}`)
+
+  const resolutionResult = await resolveDoc(didForWeb3Name)
+  if (!resolutionResult) {
+    throw 'The DID does not exist on the KILT blockchain.'
+  }
+
+  const didDetails = resolutionResult.details
+  // If no details are returned but resolutionResult is not null, the DID has been deleted.
+  // This information is present in `resolutionResult.metadata.deactivated`.
+  if (!didDetails) {
+    throw 'The DID has already been deleted.'
+  }
+
+  // Filter the endpoints by their type.
+  const didEndpoints = didDetails.getEndpoints(
+    PUBLISHED_CREDENTIAL_COLLECTION_V1_TYPE
+  )
+
+  console.log(
+    `Endpoints of type "${PUBLISHED_CREDENTIAL_COLLECTION_V1_TYPE}" for the retrieved DID:`
+  )
+  console.log(JSON.stringify(didEndpoints, null, 2))
+
+  // For demonstration, only the first endpoint and its first URL are considered.
+  const firstCredentialCollectionEndpointUrl = didEndpoints[0]?.urls[0]
+  if (!firstCredentialCollectionEndpointUrl) {
+    console.log(
+      `The DID has no service endpoints of type "${PUBLISHED_CREDENTIAL_COLLECTION_V1_TYPE}".`
+    )
+  }
+
+  // Retrieve the credentials pointed at by the endpoint. Being an IPFS endpoint, the fetching can take an arbitrarily long time or even fail if the timeout is reached.
+  // The case where the result is not a JSON should be properly handled in production settings.
+  const credentialCollection: IRequestForAttestation[] = await fetch(
+    firstCredentialCollectionEndpointUrl
+  ).then((response) => response.json())
+  console.log(`Credential collection behind the endpoint:`)
+  console.log(JSON.stringify(credentialCollection, null, 2))
```

Comment:
Why two lines for console logs? For readability?

```suggestion
  console.log(`Credential collection behind the endpoint:\n ${JSON.stringify(credentialCollection, null, 2)}`)
```

Reply:
Yeah I wanted to print the object on a new line, otherwise it ends up being displayed on the same line as the message, unless there's a way to have a new line in the same log without using `\n`?

Reply:
I don't know any other way of triggering a new line without `\n`. We can leave it like this!